chat-plugins/misc.js

```diff
@@ -195,26 +203,29 @@ exports.commands = {
 
 		for (let u in rooms) {
 			let curRoom = Rooms(rooms[u]);
-			if (curRoom.type === 'battle') {
+			if (curRoom.type === 'battle' && this.can('roomowner')) {
 				battleRooms.push('<a href="/' + curRoom.id + '" class="ilink">' + Chat.escapeHTML(curRoom.title) + '</a> (' + curRoom.userCount + ')');
 			}
 			if (curRoom.type === 'chat') {
-				if (curRoom.isPersonal) {
+				if (curRoom.isPersonal && this.can('roomowner')) {
 					groupChats.push('<a href="/' + curRoom.id + '" class="ilink">' + curRoom.id + '</a> (' + curRoom.userCount + ')');
 					continue;
 				}
 				if (curRoom.isOfficial) {
 					official.push('<a href="/' + toId(curRoom.title) + '" class="ilink">' + Chat.escapeHTML(curRoom.title) + '</a> (' + curRoom.userCount + ')');
 					continue;
 				}
-				if (curRoom.isPrivate) {
+				if (curRoom.isPrivate && this.can('roomowner')) {
 					privateRoom.push('<a href="/' + toId(curRoom.title) + '" class="ilink">' + Chat.escapeHTML(curRoom.title) + '</a> (' + curRoom.userCount + ')');
 					continue;
 				}
 			}
 			if (curRoom.type !== 'battle') nonOfficial.push('<a href="/' + toId(curRoom.title) + '" class="ilink">' + curRoom.title + '</a> (' + curRoom.userCount + ')');
 		}
+		if (!this.can('roomowner')) this.sendReplyBox(header + official.join(' ') + nonOfficial.join(' ');
```

Comment:
This is the right place and way to check permissions, but the code is a bit messy from what i can see on github. Can you try to clean this up? Something more like:

```javascript
if (!this.can('roomowner')) {
    this.sendReplyBox(header + official.join(' ') + nonOfficial.join(' ');
} else {
    this.sendReplyBox(header + official.join(' ') + nonOfficial.join(' ') + privateRoom.join(' ') + (groupChats.length > 1 ? groupChats.join(' ') : '') + (battleRooms.length > 1 ? battleRooms.join(' ') : ''));
}
```